Tests/ArmoniK.DevelopmentKit.Common.Tests/ProtoSerializerTest.cs

```diff
@@ -0,0 +1,209 @@
+  {
+    var message = Enumerable.Range(0,
+                                   5)
+                            .ToArray() as Array;
+    var serialized = ProtoSerializer.Serialize(message);
+    var result     = ProtoSerializer.Deserialize<Array>(serialized);
+    Assert.That(result,
+                Is.Not.Null);
+
+    var array = result.Cast<int>()
+                      .ToArray();
+
+    Assert.That(array.Count,
+                Is.EqualTo(5));
+    Assert.Multiple(() =>
+                    {
+                      Assert.That(array[0],
+                                  Is.EqualTo(0));
+                      Assert.That(array[1],
+                                  Is.EqualTo(1));
+                      Assert.That(array[2],
+                                  Is.EqualTo(2));
+                      Assert.That(array[3],
+                                  Is.EqualTo(3));
+                      Assert.That(array[4],
+                                  Is.EqualTo(4));
+                    });
+  }
+
+
+  [Test]
+  public void SerializeAndDeserializeArmonikPayload()
+  {
+    var message = new ArmonikPayload("methodName",
+                                     new[]
+                                     {
+                                       (byte)0x01,
+                                       (byte)0x02,
+                                     },
+                                     false);
+
+    var serialized = ProtoSerializer.Serialize(message);
+    var result     = ProtoSerializer.Deserialize<ArmonikPayload>(serialized);
+    Assert.That(result,
+                Is.Not.Null);
+
+    Assert.Multiple(() =>
+                    {
+                      Assert.That(result!.ClientPayload,
+                                  Is.Not.Null);
+                      Assert.That(result!.ClientPayload[0],
+                                  Is.EqualTo((byte)0x01));
+                      Assert.That(result!.ClientPayload[1],
+                                  Is.EqualTo((byte)0x02));
+                      Assert.That(result!.MethodName,
+                                  Is.EqualTo("methodName"));
+                      Assert.That(result!.SerializedArguments,
+                                  Is.EqualTo(false));
+                    });
+  }
+
+
+  [Test]
+  public void SerializeAndDeserializeArrayofArray()
+  {
+    var message = Enumerable.Range(0,
+                                   3)
+                            .Select(i => Enumerable.Range(0,
+                                                          2)
+                                                   .Select(i1 => $"{i},{i1}")
+                                                   .ToArray())
+                            .Append(null)
+                            .ToArray();
+    var serialized = ProtoSerializer.Serialize(message);
+    var result     = ProtoSerializer.Deserialize<object?[]>(serialized);
+    Assert.That(result,
+                Is.Not.Null);
+
+    Assert.Multiple(() =>
```

Comment:
Do we really need that ?

Reply:
It avoids a new analyzer message

Reply:
That's ugly

Reply:
It prevents the execution to stop at the first failed assert. You can have the results of all asserts in the Assert.Multiple.

Reply:
ok ! That's useful !